source/SIL.AppBuilder.Portal.Frontend/src/data/push-payload.ts

```diff
@@ -13,9 +13,14 @@ export async function pushPayload(updateStore, payload) {
 
   assignIdsToResources(resources);
 
+  console.log(resources);
+
   await updateStore(
-    q => resources.map(
-      resource => q.addRecord(resource)), { skipRemote: true });
+    q => resources.map(resource => {
+      return q[op](resource);
+    }),
+    { skipRemote: true }
+  );
```

Comment:
![payload](https://user-images.githubusercontent.com/501129/50015517-cb8c1c80-ff94-11e8-9087-96b0261f92ba.png)

Reply:
That is the payload sent to pushPayload

Reply:
PushPayload just before calling updateStore

![pushpayload](https://user-images.githubusercontent.com/501129/50015642-0b530400-ff95-11e8-8ee3-eaad0070501d.png)

Reply:
cool, that should be fine.
can you output `normalized` and `resources` as well?

Reply:
so, that guid id there shoudl hopefully match the id of that record pre-PATCH

Reply:
should match but not sure

Reply:
Normalized:

Reply:
it's falling because reviewers has no data (data:null)
![reviewers](https://user-images.githubusercontent.com/501129/50016455-3e969280-ff97-11e8-9854-c7ed44711728.png)

Reply:
ah ok, so something needs a null check. neat

Reply:
looks like the code doesn't like 

```
relationships: {
  reviewers: {
    data: null
  }
}

    SchemaConsistencyProcessor.prototype._addRelatedRecordsOps = function _addRelatedRecordsOps(record, relationshipDef, relatedRecords) {
        var _this6 = this;

        if (relatedRecords.length > 0 && relationshipDef.inverse) {
            return relatedRecords.map(function (relatedRecord) {
                return _this6._addRelationshipOp(relatedRecord, relationshipDef.inverse, record);
            });
        }
        return [];
    };
```

since relatedRecords is null .... this code breaks... I guess we need to change relationships to data: [ ]

Reply:
this code could be improved just doing this

`if (relatedRecords && relatedRecords.length > 0 && relationshipDef.inverse) { ... }`

Reply:
I think we'd want to make the change in the push-payload file, yeah?

Reply:
done